yarn-project/sequencer-client/src/tx_validator/nullifier_cache.ts

```diff
@@ -0,0 +1,25 @@
+import { MerkleTreeId, type MerkleTreeReadOperations } from '@aztec/circuit-types';
+import { type NullifierSource } from '@aztec/p2p';
+
+/**
+ * Implements a nullifier source by checking a DB and an in-memory collection.
+ * Intended for validating transactions as they are added to a block.
+ */
+export class NullifierCache implements NullifierSource {
+  nullifiers: Set<string>;
+
+  constructor(private db: MerkleTreeReadOperations) {
+    this.nullifiers = new Set();
+  }
+
+  async nullifiersExist(nullifiers: Buffer[]): Promise<boolean[]> {
+    const dbIndices = await this.db.findLeafIndices(MerkleTreeId.NULLIFIER_TREE, nullifiers);
```

Comment:
Depending on the use case here. It might make a material difference to only request from the world state only those nullifiers not found in the cache.

Reply:
Fixed in b88730116d7ac8025b7fd9e2dd64b0bc2c09bc6b